### PR TITLE
Vehicle curtain flag changes

### LINKED ISSUE #24599: Open/Close All Curtains vehicle command does not also open/close Aisle Curtains
O/S: Windows
Version: 0.C-29910-g3870791
Tiles

Open/Close All Curtains command on vehicle console only looks for the basic Curtain object that you install on windows. There are vehicle configuration where it is favorable to also have curtains on Quarterpanels to block line of sight (such as on vehicle corners next to windshields). However, Quarterpanels only allow installation of Aisle Curtains, which the command does not parse as a target.

https://github.com/CleverRaven/Cataclysm-DDA/blob/5226eca2f33979a0685aa10448fdb50451fff0b6/src/vehicle.cpp#L630-L638

### PR DESCRIPTION
**Pull request by Avatharian**
Fixes #24599

Lets Vehicle Controls toggle Aisle Curtains. Adds Curtain flag to Aisle Curtains, Window_Curtain flag to regular Curtains and updates placement logic to look for Window_Curtain to ensure curtain placement options stay the same.

### CODE REVIEW

File: src/vehicle.cpp [hunk 1]
```diff
@@ -1727,7 +1727,7 @@ bool vehicle::can_mount(int const dx, int const dy, const vpart_id &id) const
 
     // curtains must be installed on (reinforced)windshields
     // TODO: do this automatically using "location":"on_mountpoint"
-    if (part.has_flag("CURTAIN")) {
+    if (part.has_flag("WINDOW_CURTAIN")) {
```
Comment (DracoGriffin) [inline src/vehicle.cpp]: This file looks like it hasn't been astyled yet, but in the future, you'll want this line to look like this:
```cplusplus
    if( part.has_flag( "DOOR_MOTOR" ) ) {
```

or you'll end up with build failures.

If you want to try astyling it now (it shouldn't fail as this file seems to be half-astyled and half not), it should look like:

```cplusplus
    if( part.has_flag( "WINDOW_CURTAIN" ) ) {
```

File: data/json/vehicle_parts.json [hunk 2]
```diff
@@ -1130,7 +1130,7 @@
       "repair": { "skills": [ [ "mechanics", 1 ] ], "time": 2000, "using": [ [ "adhesive", 1 ] ] },
       "removal": { "skills": [ [ "mechanics", 1 ] ], "qualities": [ { "id": "SCREW", "level": 1 } ] }
     },
-    "flags": [ "OPENABLE", "OPENCLOSE_INSIDE", "OPAQUE", "CURTAIN" ],
+    "flags": [ "OPENABLE", "OPENCLOSE_INSIDE", "OPAQUE", "CURTAIN", "WINDOW_CURTAIN" ],
```
Comment (mlangsdorf) [inline data/json/vehicle_parts.json]: The info field from the WINDOW_CURTAIN flag will be displayed as part of the description, so you need to trim the description to remove the parts that are repeated here.

**Comment by Avatharian** [comment]
Ok yeah I see how that works with the description now. I'll fix up the flags and descriptions to get them to make sense all together.

File: data/json/vehicle_parts.json (hunk 2 above)
Comment (Avatharian) [reply data/json/vehicle_parts.json]: Ok yeah I see how that works with the description now. I'll fix up the flags and descriptions to get them to make sense all together.

**Pull request merged**